app/controllers/api/v1/articles_controller.rb

```diff
@@ -2,12 +2,23 @@ class Api::V1::ArticlesController < ApplicationController
   rescue_from ActiveRecord::RecordNotFound, with: :article_not_found
 
   def index
-    if params[:category]
-      articles = Article.where(category: params[:category]).paginate(page: params[:category], per_page: 4)
+    if params[:location]
+      articles = Article.where(:location => params[:location]).paginate(page: params[:page], per_page: 4)
+      if params[:category]
+        articles = articles.where(category: params[:category]).paginate(page: params[:category], per_page: 4)
+        render json: articles, each_serializer: Articles::IndexSerializer, meta: meta_attributes(articles)
+      else
+        render json: articles, each_serializer: Articles::IndexSerializer, meta: meta_attributes(articles)
+      end
     else
-      articles = Article.paginate(page: params[:page], per_page: 4)
+      if params[:category]
+        articles = Article.where(category: params[:category]).paginate(page: params[:category], per_page: 4)
+        render json: articles, each_serializer: Articles::IndexSerializer, meta: meta_attributes(articles)
+      else
+        articles = Article.paginate(page: params[:page], per_page: 4)
+        render json: articles, each_serializer: Articles::IndexSerializer, meta: meta_attributes(articles)
+      end
```

Comment:
I haven't actually test-ran this code but something like this would be a little cleaner?
```
 if params[:location] && params[:category]
   articles = Article.where(location: params[:location], category: params[:category])
 elsif params[:location]
  articles = Article.where(location: params[:location])
elsif params[:category]
  articles = Article.where(category: params[:category])
end

articles = articles.paginate(page: params[:category], per_page: 4)
render json: articles, each_serializer: Articles::IndexSerializer, meta: meta_attributes(articles)
```

Reply:
yeah, just the issue is that if the send a query something that isnt there it will return an empty string. ei. params[:location] && params[:category] will return [ ] if there is nothing in that category..

Reply:
hm okay, but that is a problem for both implementations then or is it just my refactored one?

Reply:
both, but im fixing it. BUT what you sent has helped already, getting there

Reply:
im an idiot, completely missunderstood one of olivers comments and was lookinf for a bunch of extra outcomes that iv realized dont matter at all. and yes johan your logic there is perfect